File: intellij.bazel.core/src/org/jetbrains/bazel/flow/open/OpenProjectTaskCompat.java
package org.jetbrains.bazel.flow.open;

import com.intellij.ide.impl.OpenProjectTask;
import com.intellij.ide.impl.OpenProjectTaskBuilder;
import com.intellij.ide.impl.OpenProjectTaskKt;
import kotlin.Unit;
import kotlin.jvm.functions.Function1;

final class OpenProjectTaskCompat {
  static OpenProjectTask build(Function1<? super OpenProjectTaskBuilder, Unit> buildAction) {
    return OpenProjectTaskKt.OpenProjectTask(buildAction);
  }
}

File: intellij.bazel.core/src/org/jetbrains/bazel/flow/open/OpenProjectTaskCompat.kt
package org.jetbrains.bazel.flow.open

import com.intellij.ide.impl.OpenProjectTask
import com.intellij.ide.impl.OpenProjectTaskBuilder

/**
 * Usage of [OpenProjectTask] or [OpenProjectTask.copy] leads to compatibility issues.
 * - [OpenProjectTask] - due to inlining
 * - [OpenProjectTask.copy] - due to default arguments
 */
@Suppress("FunctionName")
internal fun OpenProjectTaskCompat(buildAction: OpenProjectTaskBuilder.() -> Unit): OpenProjectTask =
  OpenProjectTaskCompat.build(buildAction)
